docs/getting-started.md

```diff
@@ -0,0 +1,325 @@
+
+
+## Configuring single-user servers
+
+Since the single-user server is an instance of `ipython notebook`,
+an entire separate multi-process application,
+there is a lot you can configure,
+and a lot of ways to express that configuration.
+
+At the JupyterHub level, you can set some values on the Spawner.
+The simplest of these is `Spawner.notebook_dir`,
+which lets you set the root directory for a user's server.
+`~` is expanded to the user's home directory.
+
+```python
+c.Spawner.notebook_dir = '~/notebooks'
+```
+
+You can also specify extra command-line arguments to the notebook server with
+
+```python
+c.Spawner.args = ['--debug', '--profile=PHYS131']
+```
+
+Since the single-user server extends the notebook server application,
+it still loads configuration from the `ipython_notebook_config.py` config file.
+Each user may have one of these files in `$HOME/.ipython/profile_default/`.
+IPython also supports loading system-wide config files from `/etc/ipython/`,
+which is the place to put configuration that you want to affect all of your users.
+
+
+- setting working directory
+- setting default page
+- /etc/ipython
+- custom Spawner
+
+## external services
+
+JupyterHub has a REST API that can be used
+
+### example: separate notebook-dir from landing url
+
+
+An example case:
+
+You are hosting JupyterHub on a single cloud server,
+using https on the standard https port, 443.
+You want to use GitHub OAuth for login,
+but need the users to exist locally on the server.
+You want users' notebooks to be served from `~/notebooks`,
+and you also want the landing page to be `~/notebooks/Welcome.ipynb`,
+instead of the directory listing page that is IPython's default.
+
+Let's start out with `jupyterhub_config.py`:
+
+```python
+c = get_config()
+
+import os
+pjoin = os.path.join
+
+runtime_dir = os.path.join('/var/run/jupyterhub')
+ssl_dir = pjoin(runtime_dir, 'ssl')
+if not os.path.exists(ssl_dir):
+    os.makedirs(ssl_dir)
+
+
+# https on :443
+c.JupyterHub.port = 443
+c.JupyterHub.ssl_key = pjoin(ssl_dir, 'ssl.key')
+c.JupyterHub.ssl_cert = pjoin(ssl_dir, 'ssl.cert')
+
+# put the JupyterHub cookie secret and state db
+# in /var/run/jupyterhub
+c.JupyterHub.cookie_secret_file = pjoin(runtime_dir, 'cookie_secret')
+c.JupyterHub.db_file = pjoin(runtime_dir, 'jupyterhub.sqlite')
+
+# use GitHub OAuthenticator for local users
+
+c.JupyterHub.authenticator_class = 'oauthenticator.LocalGitHubOAuthenticator'
+c.GitHubOAuthenticator.oauth_callback_url = os.environ['OAUTH_CALLBACK_URL']
+# create system users that don't exist yet
+c.LocalAuthenticator.create_system_users = True
+
+# specify users and admin
+c.Authenticator.whitelist = {'rgbkrk', 'minrk', 'jhamrick'}
+c.JupyterHub.admin_users = {'jhamrick', 'rgbkrk'}
+
+# start users in ~/assignments,
+# with Welcome.ipynb as the default landing page
+# this config could also be put in
+# /etc/ipython/ipython_notebook_config.py
+c.Spawner.notebook_dir = '~/assignments'
```

Comment:
I switched it to assigments to avoid confusion, since the URLs start with `/notebooks/` regardless. I updated the goals described at the top, so it's consistent.